pkg/cmd/config/delete-garden.go

```diff
@@ -0,0 +1,99 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+
+	"github.com/spf13/cobra"
+)
+
+// NewCmdConfigDeleteGarden returns a new (config) delete-garden command.
+func NewCmdConfigDeleteGarden(f util.Factory, o *DeleteGardenOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete-garden",
+		Short: "delete Garden from gardenctl configuration",
+		Long:  "Delete Garden from gardenctl configuration. E.g. \"gardenctl config delete-garden my-garden to delete my-garden",
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			suggestions, err := validGardenArgsFunction(f, args)
+			if err != nil {
+				fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+
+			return util.FilterStringsByPrefix(toComplete, suggestions), cobra.ShellCompDirectiveNoFileComp
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(f, cmd, args); err != nil {
+				return fmt.Errorf("failed to complete command options: %w", err)
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			return runDeleteGardenCommand(f, o)
+		},
```

Comment:
use `runE := base.WrapRunE(o, f)` if possible to get rid of the boilerplate code. 
Do this also for the other commands that were newly introduced

Reply:
done